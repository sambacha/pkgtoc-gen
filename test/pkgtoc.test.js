const mockConsole = require("jest-mock-console").default;
const { pkgtoc } = require("../src/pkgtoc");

const DEFAULT_OPTIONS = {
  dryRun: true,
  glob: "packages/scripts/pkgtoc/test/fixtures/*"
};

describe("pkgtoc", () => {
  beforeAll(() => mockConsole());

  describe("default", () => {
    it("should match snapshot", async () => {
      const table = await pkgtoc(DEFAULT_OPTIONS);
      expect(table).toMatchSnapshot();
    });
  });

  describe("3 columns", () => {
    it("should match snapshot", async () => {
      const table = await pkgtoc({ ...DEFAULT_OPTIONS, columns: 3 });
      expect(table).toMatchSnapshot();
    });

    it("should match snapshot w/ shields", async () => {
      const table = await pkgtoc({
        ...DEFAULT_OPTIONS,
        columns: 3,
        shields: true
      });
      expect(table).toMatchSnapshot();
    });
  });

  describe("2 columns", () => {
    it("should match snapshot", async () => {
      const table = await pkgtoc({ ...DEFAULT_OPTIONS, columns: 3 });
      expect(table).toMatchSnapshot();
    });

    it("should match snapshot w/ shields", async () => {
      const table = await pkgtoc({
        ...DEFAULT_OPTIONS,
        columns: 3,
        shields: true
      });
      expect(table).toMatchSnapshot();
    });
  });

  describe("1 columns", () => {
    it("should match snapshot", async () => {
      const table = await pkgtoc({
        ...DEFAULT_OPTIONS,
        columns: 1
      });
      expect(table).toMatchSnapshot();
    });

    it("should match snapshot with shields", async () => {
      const table = await pkgtoc({
        ...DEFAULT_OPTIONS,
        columns: 1,
        shields: true
      });
      expect(table).toMatchSnapshot();
    });
  });

  describe("colors", () => {
    it("should match snapshot", async () => {
      const table = await pkgtoc({
        ...DEFAULT_OPTIONS,
        columns: 1,
        shields: true,
        shieldsColor: "000000"
      });
      expect(table).toMatchSnapshot();
    });

    it("should match snapshot with shields", async () => {
      const table = await pkgtoc({
        ...DEFAULT_OPTIONS,
        columns: 1,
        shields: true,
        shieldsColor: "ff0000"
      });
      expect(table).toMatchSnapshot();
    });

    it("should match snapshot with auto shields", async () => {
      const table = await pkgtoc({
        ...DEFAULT_OPTIONS,
        columns: 1,
        shields: true,
        shieldsColor: "ff0000",
        shieldsAuto: true
      });
      expect(table).toMatchSnapshot();
    });
  });
});
