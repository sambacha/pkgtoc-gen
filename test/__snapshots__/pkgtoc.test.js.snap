// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pkgtoc 1 columns should match snapshot 1`] = `
"**Fixtures**

| Package                                                                                                                       |
| :---------------------------------------------------------------------------------------------------------------------------- |
| **[a-fake-package-3](packages/scripts/pkgtoc/test/fixtures/fake-package-3/)**<br />A Fake package 3 (add 'a' to show sorting) |
| **[fake-package-1](packages/scripts/pkgtoc/test/fixtures/fake-package-1/)**<br />Fake package 1                               |
| **[fake-package-2](packages/scripts/pkgtoc/test/fixtures/fake-package-2/)**<br />Fake package 2                               |
"
`;

exports[`pkgtoc 1 columns should match snapshot with shields 1`] = `
"**Fixtures**

| Package                                                                                                                                                                                                                            |
| :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| ![a-fake-package-3 npm badge)](https://img.shields.io/npm/v/a-fake-package-3.svg?colorB=cb3837)<br />**[a-fake-package-3](packages/scripts/pkgtoc/test/fixtures/fake-package-3/)**<br />A Fake package 3 (add 'a' to show sorting) |
| ![fake-package-1 npm badge)](https://img.shields.io/npm/v/fake-package-1.svg?colorB=cb3837)<br />**[fake-package-1](packages/scripts/pkgtoc/test/fixtures/fake-package-1/)**<br />Fake package 1                                   |
| ![fake-package-2 npm badge)](https://img.shields.io/npm/v/fake-package-2.svg?colorB=cb3837)<br />**[fake-package-2](packages/scripts/pkgtoc/test/fixtures/fake-package-2/)**<br />Fake package 2                                   |
"
`;

exports[`pkgtoc 2 columns should match snapshot 1`] = `
"**Fixtures**

| Package                                                                       | Description                                |
| :---------------------------------------------------------------------------- | :----------------------------------------- |
| **[a-fake-package-3](packages/scripts/pkgtoc/test/fixtures/fake-package-3/)** | A Fake package 3 (add 'a' to show sorting) |
| **[fake-package-1](packages/scripts/pkgtoc/test/fixtures/fake-package-1/)**   | Fake package 1                             |
| **[fake-package-2](packages/scripts/pkgtoc/test/fixtures/fake-package-2/)**   | Fake package 2                             |
"
`;

exports[`pkgtoc 2 columns should match snapshot w/ shields 1`] = `
"**Fixtures**

| Package                                                                       | Version                                                                                         | Description                                |
| :---------------------------------------------------------------------------- | :---------------------------------------------------------------------------------------------- | :----------------------------------------- |
| **[a-fake-package-3](packages/scripts/pkgtoc/test/fixtures/fake-package-3/)** | ![a-fake-package-3 npm badge)](https://img.shields.io/npm/v/a-fake-package-3.svg?colorB=cb3837) | A Fake package 3 (add 'a' to show sorting) |
| **[fake-package-1](packages/scripts/pkgtoc/test/fixtures/fake-package-1/)**   | ![fake-package-1 npm badge)](https://img.shields.io/npm/v/fake-package-1.svg?colorB=cb3837)     | Fake package 1                             |
| **[fake-package-2](packages/scripts/pkgtoc/test/fixtures/fake-package-2/)**   | ![fake-package-2 npm badge)](https://img.shields.io/npm/v/fake-package-2.svg?colorB=cb3837)     | Fake package 2                             |
"
`;

exports[`pkgtoc 3 columns should match snapshot 1`] = `
"**Fixtures**

| Package                                                                       | Description                                |
| :---------------------------------------------------------------------------- | :----------------------------------------- |
| **[a-fake-package-3](packages/scripts/pkgtoc/test/fixtures/fake-package-3/)** | A Fake package 3 (add 'a' to show sorting) |
| **[fake-package-1](packages/scripts/pkgtoc/test/fixtures/fake-package-1/)**   | Fake package 1                             |
| **[fake-package-2](packages/scripts/pkgtoc/test/fixtures/fake-package-2/)**   | Fake package 2                             |
"
`;

exports[`pkgtoc 3 columns should match snapshot w/ shields 1`] = `
"**Fixtures**

| Package                                                                       | Version                                                                                         | Description                                |
| :---------------------------------------------------------------------------- | :---------------------------------------------------------------------------------------------- | :----------------------------------------- |
| **[a-fake-package-3](packages/scripts/pkgtoc/test/fixtures/fake-package-3/)** | ![a-fake-package-3 npm badge)](https://img.shields.io/npm/v/a-fake-package-3.svg?colorB=cb3837) | A Fake package 3 (add 'a' to show sorting) |
| **[fake-package-1](packages/scripts/pkgtoc/test/fixtures/fake-package-1/)**   | ![fake-package-1 npm badge)](https://img.shields.io/npm/v/fake-package-1.svg?colorB=cb3837)     | Fake package 1                             |
| **[fake-package-2](packages/scripts/pkgtoc/test/fixtures/fake-package-2/)**   | ![fake-package-2 npm badge)](https://img.shields.io/npm/v/fake-package-2.svg?colorB=cb3837)     | Fake package 2                             |
"
`;

exports[`pkgtoc colors should match snapshot 1`] = `
"**Fixtures**

| Package                                                                                                                                                                                                                            |
| :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| ![a-fake-package-3 npm badge)](https://img.shields.io/npm/v/a-fake-package-3.svg?colorB=000000)<br />**[a-fake-package-3](packages/scripts/pkgtoc/test/fixtures/fake-package-3/)**<br />A Fake package 3 (add 'a' to show sorting) |
| ![fake-package-1 npm badge)](https://img.shields.io/npm/v/fake-package-1.svg?colorB=000000)<br />**[fake-package-1](packages/scripts/pkgtoc/test/fixtures/fake-package-1/)**<br />Fake package 1                                   |
| ![fake-package-2 npm badge)](https://img.shields.io/npm/v/fake-package-2.svg?colorB=000000)<br />**[fake-package-2](packages/scripts/pkgtoc/test/fixtures/fake-package-2/)**<br />Fake package 2                                   |
"
`;

exports[`pkgtoc colors should match snapshot with auto shields 1`] = `
"**Fixtures**

| Package                                                                                                                                                                                                              |
| :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| ![a-fake-package-3 npm badge)](https://img.shields.io/badge/npm-1.0.0-ff0000.svg)<br />**[a-fake-package-3](packages/scripts/pkgtoc/test/fixtures/fake-package-3/)**<br />A Fake package 3 (add 'a' to show sorting) |
| ![fake-package-1 npm badge)](https://img.shields.io/badge/npm-10.0.0-ff0000.svg)<br />**[fake-package-1](packages/scripts/pkgtoc/test/fixtures/fake-package-1/)**<br />Fake package 1                                |
| ![fake-package-2 npm badge)](https://img.shields.io/badge/npm-1.0.0-ff0000.svg)<br />**[fake-package-2](packages/scripts/pkgtoc/test/fixtures/fake-package-2/)**<br />Fake package 2                                 |
"
`;

exports[`pkgtoc colors should match snapshot with shields 1`] = `
"**Fixtures**

| Package                                                                                                                                                                                                                            |
| :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| ![a-fake-package-3 npm badge)](https://img.shields.io/npm/v/a-fake-package-3.svg?colorB=ff0000)<br />**[a-fake-package-3](packages/scripts/pkgtoc/test/fixtures/fake-package-3/)**<br />A Fake package 3 (add 'a' to show sorting) |
| ![fake-package-1 npm badge)](https://img.shields.io/npm/v/fake-package-1.svg?colorB=ff0000)<br />**[fake-package-1](packages/scripts/pkgtoc/test/fixtures/fake-package-1/)**<br />Fake package 1                                   |
| ![fake-package-2 npm badge)](https://img.shields.io/npm/v/fake-package-2.svg?colorB=ff0000)<br />**[fake-package-2](packages/scripts/pkgtoc/test/fixtures/fake-package-2/)**<br />Fake package 2                                   |
"
`;

exports[`pkgtoc default should match snapshot 1`] = `
"**Fixtures**

| Package                                                                       | Description                                |
| :---------------------------------------------------------------------------- | :----------------------------------------- |
| **[a-fake-package-3](packages/scripts/pkgtoc/test/fixtures/fake-package-3/)** | A Fake package 3 (add 'a' to show sorting) |
| **[fake-package-1](packages/scripts/pkgtoc/test/fixtures/fake-package-1/)**   | Fake package 1                             |
| **[fake-package-2](packages/scripts/pkgtoc/test/fixtures/fake-package-2/)**   | Fake package 2                             |
"
`;
